helm-charts/splunk-otel-collector/values.yaml

```diff
@@ -1360,23 +1361,60 @@ networkExplorer:
 #
 # The OpenTelemetry Operator running as a deployment with a replica count of 1.
 # It auto-instruments applications to emit telemetry data.
-# Related documentation: docs/auto-instrumentation-install.md
+# Related documentation: https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/auto-instrumentation-install.md
 # Full list of Helm value configurations: https://artifacthub.io/packages/helm/opentelemetry-helm/opentelemetry-operator?modal=values
 ################################################################################
 
 operator:
   enabled: false
-  manager:
-    autoInstrumentationImage:
+  # For more details, refer to: https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#instrumentation
+  instrumentation:
+    # Overrides for default instrumentation configurations can be specified here.
+    # Source: https://github.com/signalfx/splunk-otel-collector-chart/blob/main/helm-charts/splunk-otel-collector/templates/operator/instrumentation.yaml
+    # Rendered Default: https://github.com/signalfx/splunk-otel-collector-chart/blob/main/examples/enable-operator-and-auto-instrumentation/rendered_manifests/operator/instrumentation.yaml
+    spec:
+      # Optional "endpoint" parameter for exporting data to a specific target.
+      # By default, the endpoint will be set to the agent if it's enabled. If the agent is not enabled, the endpoint
+      # will default to the gateway, given it is enabled. If neither the agent nor the gateway is enabled, the endpoint
+      # must be overridden here.
+      # exporter:
+        # endpoint: http://$(SPLUNK_OTEL_AGENT):4317
+      propagators:
+        - tracecontext
+        - baggage
+        - b3
+      # Optional "sampler" parameter for enabling trace sampling, see: https://opentelemetry.io/docs/concepts/sdk-configuration/general-sdk-configuration/#otel_traces_sampler
+      # sampler:
+        # type: traceidratio
+        # argument: "0.95"
+      # Optional "environment variable" parameters that can configure all instrumentation libraries.
+      # env:
+        # Sets the deployment.environment tag on spans.
+        # - name: OTEL_RESOURCE_ATTRIBUTES
+        #   value: "deployment.environment=dev"
```

Comment:
This resource attribute is always set by the collector, I don't think we need to have any logic for `OTEL_RESOURCE_ATTRIBUTES` in the instrumentation. The only use case is deploying the instrumentation without an agent or gateway. I see you have some other logic to support that use case. Why do you think we need to support it? I'd say we can just fail if the agent and gateway are disabled with the instrumentation. It'll reduce a significant portion of code here. If users want instrumentation only, I think they should use the community operator instead. I'm just trying to see if we can avoid adding complexity to this chart for no apparent reason. Let me know if you disagree cc @aunshc

Reply:
From my understanding I don't think we always set deployment.environment in the collector. Let me know if I am missing something. I would support converting deployment.environment configs to be non-optional.
- https://github.com/signalfx/splunk-otel-collector-chart/blob/ba1f0d348557a91656c68ebf393881ac5f918388/docs/advanced-configuration.md?plain=1#L67
- https://github.com/signalfx/splunk-otel-collector-chart/blob/ba1f0d348557a91656c68ebf393881ac5f918388/helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl#L553C8-L553C8

Reply:
Removed these comments